mesa/experimental/jupyter_viz.py

```diff
@@ -84,15 +84,25 @@ def handle_change_model_params(name: str, value: any):
 def ModelController(model, play_interval, current_step, set_current_step):
     playing = solara.use_reactive(False)
     thread = solara.use_reactive(None)
+    # We track the previous step to detect if user resets the model via
+    # clicking the reset button or changing the parameters. If previous_step >
+    # current_step, it means a model reset happens while the simulation is
+    # still playing.
+    previous_step = solara.use_reactive(0)
```

Comment:
Have you seen the use previous hook? Might be usable here

https://solara.dev/api/use_previous

Reply:
I just tried, but with that, the simulation will always step once after the reset. Not sure why. At least this PR works, and the code is not that much different from if it were using `use_previous`.

Reply:
It's still a bit verbose. The step+1 got me curious and I was playing around a bit. 

I think a simpler solution would be to use `use_previous` on the model and then look if the previous model is equal to the current model. If it's not set playing to False and reset current step.

Reply:
It's problematic to store 2 copies of model with different specs at any given time. Consumes 2x RAM.